examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,185 @@
+
+# %%
+# Training the Ridge Regressor
+##############################
+# We use :class:`~sklearn.linear_model.Ridge`, a linear model with L2
+# regularization. We train several models, each with a different value for the
+# model parameter `alpha`. `alpha` is a positive constant that multiplies the
+# penalty term, controlling the regularization strength. For each trained model
+# we then compute the error between the true coefficients `w` the the
+# coefficients found by the model clf. We safe the identified coefficients and
+# the calculated errors for the corresponding coefficients in lists, which makes
+# it convenient for us to plot them.
 import numpy as np
 
-from sklearn.datasets import make_regression
 from sklearn.linear_model import Ridge
 from sklearn.metrics import mean_squared_error
 
 clf = Ridge()
 
-X, y, w = make_regression(
-    n_samples=10, n_features=10, coef=True, random_state=1, bias=3.5
-)
-
+# Generate values for `alpha` that are evenly distributed on a logarithmic scale
+alphas = np.logspace(-3, 4, 200)
 coefs = []
-errors = []
-
-alphas = np.logspace(-6, 6, 200)
+errors_coefs = []
 
 # Train the model with different regularisation strengths
 for a in alphas:
-    clf.set_params(alpha=a)
-    clf.fit(X, y)
+    clf.set_params(alpha=a).fit(X, y)
     coefs.append(clf.coef_)
-    errors.append(mean_squared_error(clf.coef_, w))
-
-# Display results
-plt.figure(figsize=(20, 6))
-
-plt.subplot(121)
-ax = plt.gca()
-ax.plot(alphas, coefs)
-ax.set_xscale("log")
-plt.xlabel("alpha")
-plt.ylabel("weights")
-plt.title("Ridge coefficients as a function of the regularization")
-plt.axis("tight")
-
-plt.subplot(122)
-ax = plt.gca()
-ax.plot(alphas, errors)
-ax.set_xscale("log")
-plt.xlabel("alpha")
-plt.ylabel("error")
-plt.title("Coefficient error as a function of the regularization")
-plt.axis("tight")
-
-plt.show()
+    errors_coefs.append(mean_squared_error(clf.coef_, w))
+
+# %%
+# Plotting trained Coefficients and Mean Squared Errors
+#######################################################
+# We now plot the 10 different regularized coefficients as a function of the
+# regularization parameter `alpha` where each color represents a different
+# coefficient.
+#
+# On the right-hand-side, we plot how the errors of the coefficients from the
+# estimator change as a function of regularization.
+import matplotlib.pyplot as plt
+import pandas as pd
+
+alphas = pd.Index(alphas, name="alpha")
+coefs = pd.DataFrame(coefs, index=alphas, columns=[f"Feature {i}" for i in range(10)])
+errors = pd.Series(errors_coefs, index=alphas, name="Mean squared error")
+
+fig, axs = plt.subplots(1, 2, figsize=(20, 6))
+
+coefs.plot(
+    ax=axs[0],
+    logx=True,
+    ylabel="Ridge coefficient values",
+    title="Ridge coefficients as a function of the regularization strength",
+)
+_ = errors.plot(
+    ax=axs[1],
+    logx=True,
+    ylabel="Mean squared error",
+    title="Coefficient error as a function of the regularization strength",
+)
```

Comment:
This should solve the error in the CI.

```suggestion
coefs.plot(
    ax=axs[0],
    logx=True,
    title="Ridge coefficients as a function of the regularization strength",
)
axs[0].set_ylabel("Ridge coefficient values")

errors.plot(
    ax=axs[1],
    logx=True,
    title="Coefficient error as a function of the regularization strength",
)
_ = axs[1].set_ylabel("Mean squared error")
```

We might need to do the same for `title` (if the CI fail). We would need to use `set_title`.

Reply:
I see. I will submit it with `title` as a `DataFrame.plot()` param first and we will see if it works.